src/relations/postgresql_provider.py

```diff
@@ -199,11 +200,16 @@ def update_endpoints(self, event: DatabaseRequestedEvent = None) -> None:
             database = rel_data[relation_id].get("database")
             password = secret_data.get(relation_id, {}).get("password")
 
-            # Set the read/write endpoint.
-            self.database_provides.set_endpoints(
-                relation_id,
-                f"{self.charm.primary_endpoint}:{DATABASE_PORT}",
-            )
+            try:
+                # Set the read/write endpoint.
+                self.database_provides.set_endpoints(
+                    relation_id,
+                    f"{self.charm.primary_endpoint}:{DATABASE_PORT}",
+                )
+            except PrematureDataAccessError:
+                # The relation is not yet fullin initialized
+                # Updates should be performed on a later 'update-status' cycle
```

Comment:
Better to leave a debug message in the logs.

Reply:
Done.